include/zephyr/drivers/video.h

```diff
@@ -753,6 +753,63 @@ struct video_buffer *video_buffer_alloc(size_t size);
  */
 void video_buffer_release(struct video_buffer *buf);
 
+/**
+ * @brief Search for a format that matches in a list of capabilities
+ *
+ * @param fmts The format capability list to search.
+ * @param fmt The format to find in the list.
+ * @param idx The pointer to a number of the first format that matches.
+ *
+ * @return 0 when a format is found.
+ * @return -ENOENT when no matching format is found.
+ */
+int video_format_caps_index(const struct video_format_cap *fmts, const struct video_format *fmt,
```

Comment:
All the introduced functions here are supposed for drivers uses only ?. I think it should be put in an internal header rather in the public header `video.h`.

Reply:
The users can access the format cap structure, so they could be interested in verifying: "for this particular width/height/pixelformat, what is the min/max/step?

Not sure this would ever get used that way though! Glad with putting it on a `drivers/video/video_common.h` header like often done for driver helper functions.